### DISCUSSION TITLE
Is it a personal choice to use DBus-CXX in QT applications?

### DISCUSSION THREAD

**Discussion by carlosdelfino** [dbus-cxx/dbus-cxx#77] [General]
I'm developing a Qt application to create a graphical interface for user interaction, but most of the application logic, at least the part that involves decisions and data processing, will be in another process that cannot use Qt.

Communication between the parties will be done via DBus in the first instance, if I don't hear an overload due to the DBus protocol, I can change to another RPC/IPC system.

So would it be a personal choice to use DBusCXX even in the process that uses Qt? so would you keep everything standardized, or is it better to use only in non-Qt processes?

**Comment by rm5248** [comment]
Dbus-cxx is specifically designed to be generic enough for you to use it with any framework that you choose.  The support for Qt that is provided is so that you don't need to create a new thread to listen for messages on the bus.  Specifically in terms of Qt, this also means that objects will be in the correct thread.

**Reply by rm5248** [reply]
You can also use Qt's built-in DBus support, but it is not convenient to use.

Since DBus is just a protocol, this also means that program A can use dbus-cxx, program B can use sdbus, program C can use libdbus, etc.

**Reply by carlosdelfino** [reply]
> You can also use Qt's built-in DBus support, but it is not convenient to use.

Why wouldn't it be convenient?

**Reply by rm5248** [reply]
The API is not always clear to use, and how things map between DBus objects and Qt objects is not always clear.  Since dbus-cxx makes extensive use of templated methods, it won't compile if you try to use a method of the wrong type.

**Comment by carlosdelfino** [comment]
I got it with the tips you gave me to adjust my code. I'm still not using it intensely as I intend. But I believe that it is already closer to working out.
I'm going to devote myself now to QT and SVG imaging, then I'll go back to dbus to use this service.
Thanks.
